### PR TITLE
Uncraft car parts and misc tools

### PR DESCRIPTION
**Pull request by Coolthulhu**
Added uncraft recipes for:
- All 3 alternators
- Medium and large electric motors
- Car horns and truck horns
- Polishers
- Heavy flashlights
- Concrete mixers (actually just changed existing recipe to reversible)

Small stuff (motorbike alternator and similar size) only requires a screwdriver and soldering iron, medium stuff (car alternator) requires a wrench, large stuff also metal saw. Heavy flashlight requires a metal saw and no soldering iron, because it is a flashlight sealed in a solid case.

All of the stuff above drops common electronic components. Car stuff also drops bearings. Polisher drops a small electric motor.

Car horns just doubled in usefulness.

Why does concrete mixer have a hotplate function? Who would heat up food in a concrete mixer?

### CODE REVIEW

File: data/json/recipes/recipe_deconstruction.json [hunk 1]
```diff
@@ -1446,7 +1446,191 @@
...
+        "time": 10000,
+        "reversible": true,
+        "autolearn": false,
+        "qualities" : [
+            {"id":"SCREW","level":1,"amount":1},
+            {"id":"SAW_M","level":1,"amount":1}
+        ], "components": [
+            [ [ "cable", 12 ] ],
+            [ [ "amplifier", 2 ] ],
+            [ [ "scrap", 4 ] ]
```
Comment (mugling) [inline data/json/recipes/recipe_deconstruction.json]: aluminium?

Comment (Coolthulhu) [inline data/json/recipes/recipe_deconstruction.json]: Makes sense, changed.

Though aluminum ingot is larger than the flashlight...

Comment (mugling) [inline data/json/recipes/recipe_deconstruction.json]: It's tempting to suggest adding aluminium scrap however that might be more work than you had in mind.

In any case it does provide a partial solution for #15369

**Comment by Night-Pryanik** [comment]
Aluminum scrap will be more realistic, but will need to create new item, while ingot is easier as adding right now.

**Comment by SeanMirrsen** [comment]
We could just have "scrap" and have the materials system pick up the difference. Aluminum, steel, plastic, kevlar, rubber, wood, whatnot.

On a different note, I question the presence of power supply/power converters in alternators. A device consuming power needs a PSU to normalize the input current, but a generator will generally just provide power, and lets the PSU at the receiving end handle any problems.

**Comment by Shadefang** [comment]
I don't think the larger recipes should have the metal sawing requirement. At least in my experience the only place you'd really need that is if you wanted to cut the case/etc. into smaller bits, which really isn't required for disassembly. To be fair, there are definitely some types that have to be sawed apart, but that's more common on the smaller side of things.

**Pull request merged**